sdk/src/client/api/block_builder/input_selection/remainder.rs

```diff
@@ -184,30 +190,62 @@ impl InputSelection {
             return Err(Error::MissingInputWithEd25519Address);
         };
 
-        let mut remainder_builder = BasicOutputBuilder::new_with_amount(amount_diff).with_mana(mana_diff);
+        let mut remainder_outputs = Vec::new();
 
-        remainder_builder =
-            remainder_builder.add_unlock_condition(AddressUnlockCondition::new(remainder_address.clone()));
-
-        // TODO https://github.com/iotaledger/iota-sdk/issues/1631
-        // TODO Only putting one in remainder atm so we can at least create foundries
         if let Some(native_tokens) = native_tokens_diff {
-            log::debug!("Adding {native_tokens:?} to remainder output for {remainder_address:?}");
-            remainder_builder = remainder_builder.with_native_token(*native_tokens.first().unwrap());
-        }
-
-        let remainder = remainder_builder.finish_output()?;
+            let last_index = native_tokens.len() - 1;
```

Comment:
Can this underflow?

Reply:
Couldn't, but now written as you suggested